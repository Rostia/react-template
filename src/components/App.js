import React, { PureComponent } from 'react';
import 'normalize.css';
import './App.scss';
import { SearchProvider } from 'components/common/header';
import Home from './pages/Home';

class App extends PureComponent {
  render() {
    return (
      <div>
        <SearchProvider>
          <Home />
        </SearchProvider>
      </div>
    );
  }
}

export default App;
