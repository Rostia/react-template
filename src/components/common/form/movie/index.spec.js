import '@testing-library/jest-dom';
import React from 'react';
import { render, fireEvent, screen } from 'utils/forTest';
import Form from './index';
import App from 'components/App';

const mockFilm = {
  id: 1,
  title: 'Transformer',
  release_date: '2019-06-26',
  genres: ['Action', 'Adventure'],
  poster_path: 'https://image.tmdb.org/t/p/w500/432BowXw7a4fWXSONxBaFKqvW4f.jpg',
};

describe('Form component', () => {
  it('should render', () => {
    const { asFragment } = render(<App />);

    expect(asFragment(<Form setOpen={() => { }} />)).toMatchSnapshot();
  });

  it('should render edit form', () => {
    const { asFragment } = render(<App />);

    expect(asFragment(<Form setOpen={() => { }} isEdit={true} movie={mockFilm} />)).toMatchSnapshot();
  });

  it('should contain input', () => {
    render(<Form setOpen={() => { }} />, { initialState: {} });

    expect(!!screen.getByText(/Title/i)).toEqual(true);
  });

  it("test", () => {
    render(<Form setOpen={() => { }} />, { initialState: {} });
    const btn = screen.getByText('Reset');

    fireEvent.click(btn);

    expect(btn).toBeInTheDocument();
  });

  it("Error submit", () => {
    render(<Form setOpen={() => { }} />, { initialState: {} });
    const btn = screen.getByText('Add');

    fireEvent.click(btn);

    expect(btn).toBeInTheDocument();
  });

});
