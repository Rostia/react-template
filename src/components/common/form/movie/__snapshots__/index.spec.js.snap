// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component should render 1`] = `
<DocumentFragment>
  <div>
    <main>
      <section
        class="headerComponent"
      >
        <header
          class="headerContainer"
        >
          <a
            class="logotype"
            href="/"
          >
            Netflixroulet
          </a>
          <button
            class="btn default"
            type="button"
          >
            + Add movie
          </button>
        </header>
        <div
          class="contentContainer"
        >
          <h1
            class="header"
          >
            Find Your Movie
          </h1>
          <div
            class="searchContainer"
          >
            <input
              class="searchInput"
              type="text"
              value=""
            />
            <button
              class="btn active"
              type="button"
            >
              Search
            </button>
          </div>
        </div>
      </section>
      <section
        class="filmListContainer"
      >
        <header
          class="headerContainer"
        >
          <ul
            class="filmFilterContainer"
          >
            <li
              class="filterItem"
            >
              <span
                class="link active"
                role="presentation"
              >
                All
              </span>
            </li>
            <li
              class="filterItem"
            >
              <span
                class="link"
                role="presentation"
              >
                Dramma
              </span>
            </li>
            <li
              class="filterItem"
            >
              <span
                class="link"
                role="presentation"
              >
                Action
              </span>
            </li>
            <li
              class="filterItem"
            >
              <span
                class="link"
                role="presentation"
              >
                Oscar
              </span>
            </li>
            <li
              class="filterItem"
            >
              <span
                class="link"
                role="presentation"
              >
                Adventure
              </span>
            </li>
          </ul>
          <div
            class="filmSortContainer"
          >
            <span
              class="title"
            >
              Sort: 
            </span>
            <button
              class="sortBtn"
              type="button"
            >
              Release date
            </button>
          </div>
        </header>
        <p
          class="result"
        >
          <span>
            0
          </span>
           movies count
        </p>
        <main
          class="contentContainer"
        >
          <p
            class="noResult"
          >
            No Movie Found
          </p>
        </main>
      </section>
      <footer
        class="footerCompanent"
      >
        <a
          class="logotype"
          href="/"
        >
          Netflixroulet
        </a>
      </footer>
    </main>
  </div>
</DocumentFragment>
`;

exports[`Form component should render edit form 1`] = `
<DocumentFragment>
  <div>
    <main>
      <section
        class="headerComponent"
      >
        <header
          class="headerContainer"
        >
          <a
            class="logotype"
            href="/"
          >
            Netflixroulet
          </a>
          <button
            class="btn default"
            type="button"
          >
            + Add movie
          </button>
        </header>
        <div
          class="contentContainer"
        >
          <h1
            class="header"
          >
            Find Your Movie
          </h1>
          <div
            class="searchContainer"
          >
            <input
              class="searchInput"
              type="text"
              value=""
            />
            <button
              class="btn active"
              type="button"
            >
              Search
            </button>
          </div>
        </div>
      </section>
      <section
        class="filmListContainer"
      >
        <header
          class="headerContainer"
        >
          <ul
            class="filmFilterContainer"
          >
            <li
              class="filterItem"
            >
              <span
                class="link active"
                role="presentation"
              >
                All
              </span>
            </li>
            <li
              class="filterItem"
            >
              <span
                class="link"
                role="presentation"
              >
                Dramma
              </span>
            </li>
            <li
              class="filterItem"
            >
              <span
                class="link"
                role="presentation"
              >
                Action
              </span>
            </li>
            <li
              class="filterItem"
            >
              <span
                class="link"
                role="presentation"
              >
                Oscar
              </span>
            </li>
            <li
              class="filterItem"
            >
              <span
                class="link"
                role="presentation"
              >
                Adventure
              </span>
            </li>
          </ul>
          <div
            class="filmSortContainer"
          >
            <span
              class="title"
            >
              Sort: 
            </span>
            <button
              class="sortBtn"
              type="button"
            >
              Release date
            </button>
          </div>
        </header>
        <p
          class="result"
        >
          <span>
            0
          </span>
           movies count
        </p>
        <main
          class="contentContainer"
        >
          <p
            class="noResult"
          >
            No Movie Found
          </p>
        </main>
      </section>
      <footer
        class="footerCompanent"
      >
        <a
          class="logotype"
          href="/"
        >
          Netflixroulet
        </a>
      </footer>
    </main>
  </div>
</DocumentFragment>
`;
